src/spi.rs

```diff
@@ -1665,136 +1711,157 @@ fn set_keep_cs_active(transaction: &mut spi_transaction_t, _keep_cs_active: bool
 
 fn spi_transmit(
     handle: spi_device_handle_t,
-    transactions: impl Iterator<Item = spi_transaction_t>,
+    transactions: impl Iterator<Item = (spi_transaction_t, u32)>,
     polling: bool,
     queue_size: usize,
 ) -> Result<(), EspError> {
+    let delay_impl = delay::Delay::new_default();
+
     if polling {
-        for mut transaction in transactions {
+        for (mut transaction, delay) in transactions {
```

Comment:
I think the approach with an empty transaction does not work here, because in the past it behaved wiredly. I think we need seperation here. E.G all normal operations can call spi_device_polling_transmit, but if we have an delay operation just delay till the next operation without insertion of a dummy one.

Reply:
The empty transaction is **only** needed to model delays before/after the sequence of operations. In fact, without an "empty" transaction, there is no way at all to model delays at the beginning and at the end that keep the CS raised.

But then again, if these are not useful and/or keeping the CS raised during those is not necessary, that's an easy fix.

Reply:
Resolved in the way we agreed in the matrix chat:
* initial/final delays are OK with the software-based CS management
* initial/final delays are also OK with the hardware-based CS management but (a) the user has to opt-in into those, (b) they are - unfortunately - not waited-on during the period when the CS is active